### PR TITLE
Docs: Concise intro

### PR DESCRIPTION
**Pull request by dev2049**
Make doc intro concise

https://python.langchain.com/en/dev2049-concise_get_started/

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -67,38 +52,38 @@ For each module LangChain provides standard, extendable interfaces. LangChain al
 
    ./modules/models.rst
    ./modules/prompts.rst
-   ./modules/memory.md
-   ./modules/indexes.md
-   ./modules/chains.md
-   ./modules/agents.md
+   ./modules/memory.rst
```
Comment (eyurtsev) [inline docs/index.rst:55]: I seem to recall that we don't need to include the suffix. If that's the case, we might want to drop it so we can replace `rst` files with `md` files in the future without having to make changes in too many places.

**Review by eyurtsev — APPROVED** [review]

**Review by leo-gan — COMMENTED** [review]
It is a great text cleaning. It makes the Index page much simpler to comprehend. I like it.
I have several questions in the comments.

File: docs/index.rst [hunk 2]
```diff
@@ -1,28 +1,19 @@
 Welcome to LangChain
 ==========================
 
-| **LangChain** is a framework for developing applications powered by language models. We believe that the most powerful and differentiated applications will not only call out to a language model, but will also be:
-1. *Data-aware*: connect a language model to other sources of data
-2. *Agentic*: allow a language model to interact with its environment
+Overview
```
[the remark this answers is not in the document]
Comment (dev2049) [reply docs/index.rst:4]: don't see an Overview here, but also don't feel strongly, happy to remove! https://python.langchain.com/en/dev2049-concise_get_started/

[the remark this answers is not in the document]
Comment (leo-gan) [reply docs/index.rst:4]: Hmmm... It is not creating an Overview on the navbar.
Then I'm OK with it.

**Pull request closed without merging**